keyboards/planck/keymaps/floodedcodeboy/keymap.c

```diff
@@ -0,0 +1,305 @@
+    {6, 1, HSV_ORANGE}
+);
+
+// MOD KEY COLOUR LAYERS
+const rgblight_segment_t PROGMEM rbg_layer_mod1[] = RGBLIGHT_LAYER_SEGMENTS(
+    {8, 1, HSV_MAGENTA}
+);
+const rgblight_segment_t PROGMEM rgb_layer_mod2[] = RGBLIGHT_LAYER_SEGMENTS(
+    {1, 1, HSV_PURPLE}
+);
+const rgblight_segment_t PROGMEM rgb_layer_mod3[] = RGBLIGHT_LAYER_SEGMENTS(
+    {2, 1, HSV_AZURE}
+);
+
+const rgblight_segment_t* const PROGMEM _rgb_layers[] = RGBLIGHT_LAYERS_LIST(
+    rgb_layer_base,
+    rgb_layer_raise,    // Overrides caps lock layer
+    rgb_layer_lower,    // Overrides other layers
+    rgb_layer_adj,     // Overrides other layers
+    rgb_layer_shift,
+    rbg_layer_mod1,
+    rgb_layer_mod2,
+    rgb_layer_mod3
+);
+
+void keyboard_post_init_user(void) {
+    rgblight_layers = _rgb_layers;
+}
+
+#ifdef AUDIO_ENABLE
+    float plover_song[][2]    = SONG(PLOVER_SOUND);
+    float plover_gb_song[][2] = SONG(PLOVER_GOODBYE_SOUND);
+#endif
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case LOWER:
+            rgblight_set_layer_state(1, record->event.pressed);
+            break;
+        case RAISE:
+            rgblight_set_layer_state(2, record->event.pressed);
+            break;
+        case _ADJUST:
+            rgblight_set_layer_state(3, record->event.pressed);
+            break;
+        case KC_LSFT:
+            rgblight_set_layer_state(4, record->event.pressed);
+            break;
+        case KC_LCTL:
+            rgblight_set_layer_state(5, record->event.pressed);
+            break;
+        case KC_LALT:
+            rgblight_set_layer_state(6, record->event.pressed);
+            break;
+        case KC_LGUI:
+            rgblight_set_layer_state(7, record->event.pressed);
+            break;
+        default:
+            // silence
+            break;
+    }
+    return true;
+}
+
+bool muse_mode = false;
+uint8_t last_muse_note = 0;
+uint16_t muse_counter = 0;
+uint8_t muse_offset = 70;
+uint16_t muse_tempo = 50;
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (muse_mode) {
+        if (IS_LAYER_ON(_RAISE)) {
+            if (clockwise) {
+                muse_offset++;
+            } else {
+                muse_offset--;
+            }
+        } else {
+            if (clockwise) {
+                muse_tempo += 1;
+            } else {
+                muse_tempo -= 1;
+            }
+        }
+    } else {
+        if (clockwise) {
+            #ifdef MOUSEKEY_ENABLE
+                tap_code(KC_MS_WH_DOWN);
+            #else
+                tap_code(KC_PGDN);
+            #endif
+        } else {
+            #ifdef MOUSEKEY_ENABLE
+                tap_code(KC_MS_WH_UP);
+            #else
+                tap_code(KC_PGUP);
+            #endif
+        }
+    }
+    return true;
+}
```

Comment:
You may want to use encoder maps here.  Though, if you do and want to still use the muse functionality, you'd need to add custom keycodes to handle the muse functions.